### DISCUSSION TITLE
PDF Actions & Custom Keystroke Script

### DISCUSSION THREAD

**Discussion by hari-narendran** [stephanrauh/ngx-extended-pdf-viewer#1320] [Q&A]
Hi
I came across 2 issues related to text box and checkbox that I need help with. 
I have attached a sample PDF with the 2 scenarios

_Scenario 1 Text Field :_
Text filed has Custom Keystroke Script ( Text Filed Properties -> Custom Keystroke Script ) set to 
`if ( event.fieldFull )
{
this.getField("field_2").setFocus();
}`
With this script, text is supposed to overflow from field_1 to filed_2 (works in acrobat). But the overflow doesn't happen extended-pdf-viewer.

_Scenario 2 Check Box:_
Check box has an Action set to trigger on Mouse Up. Action is to show/hide a field. IN extended-pdf-viewer, show/hide of the filed is not working. 

Thanks in advance. 
[testpdf.pdf](https://github.com/stephanrauh/ngx-extended-pdf-viewer/files/8525128/testpdf.pdf)

**Comment by stephanrauh** [comment]
Did you already check if the features work in the showcase of the base library? https://mozilla.github.io/pdf.js/web/viewer.html

I didn't implement the JavaScript engine myself, so I'm afraid I can't help you. But if you can reproduce the same bug at the base library, you can file an issue there, and once the Mozilla team has solved it, I can add it to ngx-extended-pdf-viewer.

**Reply by hari-narendran** [reply]
I checked the same pdf with pdf.js base library now and I was able to reproduce the issue. I will file an issue there and leave a comment here with the update. 
Thanks Stephan.

**Reply by stephanrauh** [reply]
@calixteman Here's a bug report that might be interesting to you.

**Reply by hari-narendran** [reply]
Sorry that link takes to developer profile. Is there a direct link to the bug ?
Thank you

**Reply by stephanrauh** [reply]
Oops, sorry - I wanted to make the developer who is most likely to look at your bug aware that you're about to file a bug. I didn't report the bug myself, and I didn't want to confuse you. :)

**Comment by calixteman** [comment]
The `fieldFull` property exists by itself in the `event`:
https://github.com/mozilla/pdf.js/blob/75ac897284fe5da7f56ce894f4cba31270931365/src/scripting_api/event.js#L21
but it isn't used:
https://github.com/mozilla/pdf.js/blob/75ac897284fe5da7f56ce894f4cba31270931365/src/display/annotation_layer.js#L1037
or
https://github.com/mozilla/pdf.js/blob/75ac897284fe5da7f56ce894f4cba31270931365/src/display/annotation_layer.js#L1058

Please file a bug upstream: https://github.com/mozilla/pdf.js/
